includes/admin/class-sensei-extensions.php

```diff
@@ -119,16 +119,38 @@ private function add_installed_extensions_properties( $extensions ) {
 
 		$installed_plugins = get_plugins();
 
-		// Includes installed version, and whether it has update.
+		$wccom_connected     = false;
+		$wccom_subscriptions = [];
+
+		if ( class_exists( 'WC_Helper_Options' ) ) {
+			$auth            = WC_Helper_Options::get( 'auth' );
+			$wccom_connected = ! empty( $auth['access_token'] );
+
+			$wccom_subscriptions = WC_Helper::get_subscriptions();
+		}
+
+		// Includes installed version, whether it has update and WC.com metadata.
 		$extensions = array_map(
-			function( $extension ) use ( $installed_plugins ) {
+			function( $extension ) use ( $installed_plugins, $wccom_connected, $wccom_subscriptions ) {
 				$extension->is_installed = isset( $installed_plugins[ $extension->plugin_file ] );
 
 				if ( $extension->is_installed ) {
 					$extension->installed_version = $installed_plugins[ $extension->plugin_file ]['Version'];
 					$extension->has_update        = isset( $extension->version ) && version_compare( $extension->version, $extension->installed_version, '>' );
 				}
 
+				if ( isset( $extension->wccom_product_id ) ) {
+					$extension->wccom_connected = $wccom_connected;
```

Comment:
I think since it's more generic, instead of related to the extensions, we could handle this generically. I think you made it like this to avoid changing the API structure, right?

Maybe we could expose it to the client with `wp_localize_script`, WDYT?

Reply:
> I think since it's more generic, instead of related to the extensions, we could handle this generically. I think you made it like this to avoid changing the API structure, right?

Yeah I just found it a bit excessive to create an extra level in the responce just for this single field. But truth is that this way it will be more expandable later.

> Maybe we could expose it to the client with `wp_localize_script`, WDYT?

I prefer returning all data in a single place. I'll add it to the respone.

Reply:
Fixed in 7a527835cbed6a84e6d6b0c61a327da180b212e0